misc/dbt-materialize/dbt/include/materialize/macros/catalog.sql

```diff
@@ -66,3 +66,32 @@
   {{ return(load_result('catalog').table) }}
 
 {%- endmacro %}
+
+{% macro postgres__list_relations_without_caching(schema_relation) %}
+  {% call statement('list_relations_without_caching', fetch_result=True) -%}
+    select
+      '{{ schema_relation.database }}' as database,
+      tablename as name,
+      schemaname as schema,
+      'table' as type
+    from pg_tables
+    where schemaname ilike '{{ schema_relation.schema }}'
+    union all
+    select
+      '{{ schema_relation.database }}' as database,
+      viewname as name,
+      schemaname as schema,
+      'view' as type
+    from pg_views
+    where schemaname ilike '{{ schema_relation.schema }}'
+    union all
+    select
+      '{{ schema_relation.database }}' as database,
+      matviewname as name,
+      schemaname as schema,
+      'materializedview' as type
+    from pg_matviews
+    where schemaname ilike '{{ schema_relation.schema }}'
+  {% endcall %}
+  {{ return(load_result('list_relations_without_caching').table) }}
+{% endmacro %}
```

Comment:
Do you recall what required adding this back? I added it in c770e89d42ffdd3b0299ef0e3601866fae979afa, but then it got removed in 7f56f812bfbd9b032ddc259ba06de92355ad3c35, and now I'm confused. Best guess: until you added `old_relation = adapter.get_relation(...)` to some of the macros below, no one noticed when `adapter.get_relation` suddenly stopped working right?

Reply:
Note that this macros is not the same as the one from the Postgres adapter, but adds the third select from `pg_matviews`. I needed to add this because the implementation from the Postgres adapter only looks for tables and views (in `pg_tables` and `pg_views`), but not materialized views (in `pg_matviews`). That was fine until now because our adapter only created regular views (with and without indexes) so all relations were picked up. But after materialized views became a separate catalog type, the Postgres macro wouldn't find them anymore, so `adapter.get_relation` would just return `None` for all materializations but "view".

So I think everything worked correctly before, but I definitely needed to make this change to make the tests pass in this PR.

Reply:
Ahh, that makes sense. So it all worked previously with the default Postgres implementation; now that the adapter creates matviews we need a modified version of this macro that looks for matviews.